hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -181,126 +181,73 @@ public Pipeline createPipeline(ReplicationType type, ReplicationFactor factor,
   @Override
   public Pipeline getPipeline(PipelineID pipelineID)
       throws PipelineNotFoundException {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipeline(pipelineID);
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipeline(pipelineID);
   }
 
   @Override
   public boolean containsPipeline(PipelineID pipelineID) {
-    lock.readLock().lock();
     try {
       getPipeline(pipelineID);
       return true;
     } catch (PipelineNotFoundException e) {
       return false;
-    } finally {
-      lock.readLock().unlock();
     }
   }
 
   @Override
   public List<Pipeline> getPipelines() {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipelines();
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipelines();
   }
 
   @Override
   public List<Pipeline> getPipelines(ReplicationType type) {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipelines(type);
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipelines(type);
   }
 
   @Override
   public List<Pipeline> getPipelines(ReplicationType type,
                               ReplicationFactor factor) {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipelines(type, factor);
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipelines(type, factor);
   }
 
   @Override
   public List<Pipeline> getPipelines(ReplicationType type,
                               Pipeline.PipelineState state) {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipelines(type, state);
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipelines(type, state);
   }
 
   @Override
   public List<Pipeline> getPipelines(ReplicationType type,
                               ReplicationFactor factor,
                               Pipeline.PipelineState state) {
-    lock.readLock().lock();
-    try {
-      return stateManager.getPipelines(type, factor, state);
-    } finally {
-      lock.readLock().unlock();
-    }
+    return stateManager.getPipelines(type, factor, state);
   }
 
   @Override
   public List<Pipeline> getPipelines(
       ReplicationType type, ReplicationFactor factor,
       Pipeline.PipelineState state, Collection<DatanodeDetails> excludeDns,
       Collection<PipelineID> excludePipelines) {
-    lock.readLock().lock();
-    try {
-      return stateManager
+    return stateManager
```

Comment:
so the dead lock solution seems is removing the read write lock? Is there an explanation why we can remove locks here?

Reply:
I will create a jira for this issue, can give the explanation of this solution there.